testament/testament.nim

```diff
@@ -897,28 +1093,65 @@ func extraOptions(run: TestRun): string =
   else:
     ""
 
-proc testSpecHelper(r: var TResults, run: var TestRun) =
+proc exeFile(
+    testRun: TestRun, specFilePath: string, rootDir: string): string =
+  ## Get name of the executable file for the test run
+  let
+    target = testRun.target
+    isJsTarget = target == targetJs
+    (dirPart, specName, _) = splitFile(specFilePath)
+    matrixEntry = testRun.matrixEntry
+    exeName =
+      (if matrixEntry == noMatrixEntry:
+        @[specName, target.cmd]
+      else:
+        @[specName, $matrixEntry, target.cmd]).join("_")
+    exeExt =
+      if isJsTarget:
+        "js"
+      else:
+        ExeExt
+
+    fileDir = if dirPart.isAbsolute():
+                dirPart
+
+              else:
```

Comment:
No spacing/breaking apart branches like this the if and the else go together forming one expression for the assignment, there must not be any visual separation as it's not communicating the correct thing.